File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>


int main(void)
{
    string text = get_string("Text: ");

    float letters, words, sentences;
    letters = words = sentences = 0.0;
    int index;
    int n = strlen(text) + 1;

    // iterate through text[i]
    for (int i = 0; i < n; i++)
    {
        //count letters
        if (isalpha(text[i]))
        {
            letters++;
        }

        //count words
        if (text[i] == ' ' || text[i] == '\n' || text[i] == '\0')
        {
            words++;
        }

        //count sentences
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    index = round(5.88 * (letters / words) - 29.6 * (sentences / words) - 15.8);

    //print Grade
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
L = 0
W = 1
S = 0

while True:
    try:
        text = str(input("Text: "))
    except ValueError:
        continue
    break

# count letters
for letter in text:
    if letter.isalpha():
        L += 1

# count words
for word in text:
    if word == ' ' or word == '\n':
        W += 1

# count sentences
for sentence in text:
    if sentence == '.' or sentence == '?' or sentence == '!':
        S += 1

index = round(5.88 * (L/W) - 29.6 * (S/W) - 15.8)

# print Grade
if index >= 16:
    print("Grade 16+")

elif index < 1:
    print("Before Grade 1")

else:
    print("Grade", index)